install.c

```diff
@@ -509,100 +586,102 @@ int inst_choose_partition(char **partition, int swap, char *txt_menu, char *txt_
       rc = 0;
     }
 
-    if(i == item_mk_part) {
-      do {
-        i = dia_list("create a swap partition", 36, NULL, items2, last_item2, align_left);
-        if(i > 0 && values2[i - 1]) {
-          str_copy(&last_part, values2[i - 1]);
-          dev = long_dev(values2[i - 1]);
-          sprintf(buf, "Warning: all data on %s will be deleted!", dev);
-          j = dia_contabort(buf, NO);
-          if(j == YES) {
-            sprintf(buf, "/sbin/mkswap %s", dev);
-            if(!lxrc_run(buf)) {
-              log_info("swapon %s\n", dev);
-              if(swapon(dev, 0)) {
-                log_info("swapon: ");
-                perror_info(dev);
-                dia_message("Error activating swap space.", MSGTYPE_ERROR);
+    if((flags & ICP_SWAP)) {
```

Comment:
In all other cases, the old 'if' condition was if (swap), now it was something completely different - please double-check if this is really what you want